occurrence-ws/src/main/java/org/gbif/occurrence/ws/resources/DownloadResource.java

```diff
@@ -106,9 +109,47 @@ public Response oozieCallback(@QueryParam("job_id") String jobId, @QueryParam("s
     return Response.ok().build();
   }
 
+  @GET
+  @Path("sql/validate")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response validateSQL(@QueryParam("sql") String sqlquery) {
+    LOG.debug("Received validation request for sql query [{}]",sqlquery);
+    HiveSQL.Validate.Result result =  new HiveSQL.Validate().apply(sqlquery);
+    return Response.ok().type(MediaType.APPLICATION_JSON).entity(result.toString()).build();
+  }
+  
+  @GET
+  @Path("sql/describe")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response describeSQL() {
+    LOG.debug("Received describe request for sql ");
+    if (describeResponseCache.isEmpty())  { describeResponseCache = new HiveSQL.Execute().describe("occurrence_hdfs"); }
+    return Response.ok().type(MediaType.APPLICATION_JSON).entity(describeResponseCache).build();
+  }
+  
+  @POST
+  @Validate
+  @Path("sql")
+  public String startDownload(@Valid SqlDownloadRequest request, @Context SecurityContext security) {
+    LOG.debug("Download: [{}]", request);
+
+    // assert authenticated user is the same as in download
+    assertLoginMatches(request, security);
+
+    HiveSQL.Validate.Result result = new HiveSQL.Validate().apply(request.getSql());
```

Comment:
maybe if we move this validation to the DownloadReqestServiceImpl we can avoid the artificial field sqlHeader

Reply:
done.